x/evm/keeper/state_transition.go

```diff
@@ -158,6 +155,10 @@ func (k *Keeper) ApplyTransaction(tx *ethtypes.Transaction) (*types.MsgEthereumT
 	k.SetTxHashTransient(txHash)
 	k.IncreaseTxIndexTransient()
 
+	if !k.ctxStack.IsClean() {
+		panic("context stack shouldn't be dirty before apply message")
+	}
```

Comment:
this should reset the stack IMO, unless there is a good reason to panic?

Reply:
Normally, `WithContext` will be called somewhere before this which does the reset, here it do an assertion on the precondition?